std/path.d

```diff
@@ -4032,20 +4034,21 @@ version(unittest) import std.process : environment;
         if (oldHome !is null) environment["HOME"] = oldHome;
         else environment.remove("HOME");
 
-        // Test user expansion for root, no /root on Android
-        version (OSX)
-        {
-            assert(expandTilde("~root") == "/var/root", expandTilde("~root"));
-            assert(expandTilde("~root/") == "/var/root/", expandTilde("~root/"));
-        }
-        else version (Android)
+        version (Android)
         {
         }
         else
         {
-            assert(expandTilde("~root") == "/root", expandTilde("~root"));
-            assert(expandTilde("~root/") == "/root/", expandTilde("~root/"));
+            immutable user = environment.get("USER");
+            immutable tildeUser = "~" ~ user;
+            immutable echoCmd = executeShell("echo " ~ tildeUser);
+            immutable pathFromEcho = echoCmd.output.strip();
+            immutable expTildeUser = expandTilde(tildeUser);
+            immutable expTildeUserSlash = expandTilde(tildeUser ~ "/");
```

Comment:
We usually put the expression to validate directly before its respective `assert`, i.e. we would have moved this `asser` one line below.

Reply:
Moved one assert up, I guess that's what you meant.